internal/clients/init.go

```diff
@@ -155,76 +155,14 @@ func checkServiceAvailableViaRegistry(serviceId string) bool {
 
 func initializeClients(ctx context.Context, waitGroup *sync.WaitGroup) {
 	// initialize Core Metadata clients
-	common.AddressableClient = metadata.NewAddressableClient(
-		urlclient.NewMetadata(
-			ctx,
-			common.RegistryClient,
-			waitGroup,
-			clients.ApiAddressableRoute,
-		),
-	)
-
-	common.DeviceClient = metadata.NewDeviceClient(
-		urlclient.NewMetadata(
-			ctx,
-			common.RegistryClient,
-			waitGroup,
-			clients.ApiDeviceRoute,
-		),
-	)
-
-	common.DeviceServiceClient = metadata.NewDeviceServiceClient(
-		urlclient.NewMetadata(
-			ctx,
-			common.RegistryClient,
-			waitGroup,
-			clients.ApiDeviceServiceRoute,
-		),
-	)
-
-	common.DeviceProfileClient = metadata.NewDeviceProfileClient(
-		urlclient.NewMetadata(
-			ctx,
-			common.RegistryClient,
-			waitGroup,
-			clients.ApiDeviceProfileRoute,
-		),
-	)
-
-	common.MetadataGeneralClient = general.NewGeneralClient(
-		urlclient.NewMetadata(
-			ctx,
-			common.RegistryClient,
-			waitGroup,
-			"",
-		),
-	)
-
-	common.ProvisionWatcherClient = metadata.NewProvisionWatcherClient(
-		urlclient.NewMetadata(
-			ctx,
-			common.RegistryClient,
-			waitGroup,
-			clients.ApiProvisionWatcherRoute,
-		),
-	)
+	common.AddressableClient = metadata.NewAddressableClient(local.New(common.CurrentConfig.Clients[common.ClientMetadata].Url() + clients.ApiAddressableRoute))
+	common.DeviceClient = metadata.NewDeviceClient(local.New(common.CurrentConfig.Clients[common.ClientMetadata].Url() + clients.ApiDeviceRoute))
+	common.DeviceServiceClient = metadata.NewDeviceServiceClient(local.New(common.CurrentConfig.Clients[common.ClientMetadata].Url() + clients.ApiDeviceServiceRoute))
+	common.DeviceProfileClient = metadata.NewDeviceProfileClient(local.New(common.CurrentConfig.Clients[common.ClientMetadata].Url() + clients.ApiDeviceProfileRoute))
+	common.MetadataGeneralClient = general.NewGeneralClient(local.New(common.CurrentConfig.Clients[common.ClientMetadata].Url()))
+	common.ProvisionWatcherClient = metadata.NewProvisionWatcherClient(local.New(common.CurrentConfig.Clients[common.ClientMetadata].Url() + clients.ApiProvisionWatcherRoute))
 
 	// initialize Core Data clients
-	common.EventClient = coredata.NewEventClient(
```

Comment:
LGTM. Might be another issue but I found out the other day that the coredata config entry name is different between device services and edgex-go services. (in device service it is `Clients.Data` but in edgex-go it is `Clients.Coredata`). I guess it should be consistent otherwise the docker-compose ENV vars would only override partial of them.

Reply:
@hahattan , yes we are aware of this. I Geneva and Nightly compose we have overrides for both forms. Changing the key ten to CoreData for device services would be a backward breaking change, so we have to wait until v2.0.0 to resolve this inconsistency. I will add an issue to address this when we know we are release v2.0.0. THX